fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/MetricUnionAvailabilitySpec.groovy

```diff
@@ -0,0 +1,233 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability
+
+import com.google.common.collect.Sets
+import com.yahoo.bard.webservice.data.config.names.TableName
+import com.yahoo.bard.webservice.data.metric.MetricColumn
+import com.yahoo.bard.webservice.table.PhysicalTable
+import com.yahoo.bard.webservice.table.PhysicalTableSchema
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+
+import org.joda.time.Interval
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+/**
+ * Test for metric union availability behavior.
+ */
+class MetricUnionAvailabilitySpec extends Specification {
+    Availability availability1
+    Availability availability2
+
+    MetricColumn metricColumn1
+    MetricColumn metricColumn2
+
+    PhysicalTableSchema schema1
+    PhysicalTableSchema schema2
+
+    PhysicalTable physicalTable1
+    PhysicalTable physicalTable2
+
+    Set<PhysicalTable> physicalTables
+
+    MetricUnionAvailability metricUnionAvailability
+
+    def setup() {
+        availability1 = Mock(Availability)
+        availability2 = Mock(Availability)
+
+        availability1.getDataSourceNames() >> Sets.newHashSet(TableName.of('source1'))
+        availability2.getDataSourceNames() >> Sets.newHashSet(TableName.of('source2'))
```

Comment:
@cdeszaq This might be a groovy bug. If I use `[TableName.of('source1')] as Set` I get `NullPointerException: Cannot invoke method rightShift() on null object`